app/javascript/packs/modal.js

```diff
@@ -0,0 +1,36 @@
+document.addEventListener('DOMContentLoaded', function(){
```

Comment:
Rails uses turbolinks out-of-the-box for adding some functionality to our web-apps.

One of the things turbolinks does is to change how pages are loaded. I'd recommend reading the turbolinks README for understanding the lifecycle it provides.

That said, I'd not recommend listening events like `DOMContentLoaded`, due the page-loading-process turbolinks overrides. It's better listening the `turbolinks:load` event this way:

```javascript 
document.addEventListener('turbolinks:load',...)
```
For more info: https://guides.rubyonrails.org/working_with_javascript_in_rails.html#page-change-events